processor/routingprocessor/metrics.go

```diff
@@ -113,6 +130,7 @@ func (p *metricsProcessor) route(ctx context.Context, tm pmetric.Metrics) error
 					return err
 				}
 				p.group("", groups, p.router.defaultExporters, rmetrics)
+				p.recordNonRoutedForResourceMetrics(ctx, "", rmetrics)
```

Comment:
Probably `key` should be passed as a second argument. It is passed in `p. recordNonRoutedResourceLogs` and `p. recordNonRoutedResourceSpans`.
```suggestion
				p.recordNonRoutedForResourceMetrics(ctx, key, rmetrics)
```